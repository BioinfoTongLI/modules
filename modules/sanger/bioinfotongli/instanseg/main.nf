process BIOINFOTONGLI_INSTANSEG {
    tag "$meta.id"
    label 'process_medium'
    label "gpu"

    // conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'quay.io/cellgeni/instanseg:v0.0.1':
        'quay.io/cellgeni/instanseg:v0.0.1' }"

    input:
    tuple val(meta), val(x_min), val(y_min), val(x_max), val(y_max), path(img)

    output:
    tuple val(meta), path("${prefix}.wkt"), emit: wkts
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    instanseg_helper.py run \\
        -image-path ${img} \\
        -x-min ${x_min} \\
        -y-min ${y_min} \\
        -x-max ${x_max} \\
        -y-max ${y_max} \\
        -output ${prefix}.wkt \\
        $args \\

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bioinfotongli: \$(instanseg_helper.py version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.wkt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bioinfotongli: \$(instanseg_helper.py version)
    END_VERSIONS
    """
}
