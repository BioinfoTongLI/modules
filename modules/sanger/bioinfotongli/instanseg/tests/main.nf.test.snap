{
    "mouse_heart - tiff - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6587b1fbffd0ce3ddd72cb0dea70eeef"
                ],
                "versions": [
                    "versions.yml:md5,6587b1fbffd0ce3ddd72cb0dea70eeef"
                ],
                "wkts": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-06T17:20:48.858396339"
    },
    "mouse_heart - tiff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart.wkt:md5,deacacbf4602021995b9a5152e4eb8bd"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6587b1fbffd0ce3ddd72cb0dea70eeef"
                ],
                "versions": [
                    "versions.yml:md5,6587b1fbffd0ce3ddd72cb0dea70eeef"
                ],
                "wkts": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart.wkt:md5,deacacbf4602021995b9a5152e4eb8bd"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-06T17:20:39.953890219"
    }
}